src/perform_step/extrapolation_perform_step.jl

```diff
@@ -369,6 +395,33 @@ function perform_step!(integrator,cache::ImplicitEulerExtrapolationConstantCache
         T[i,j] = ((2^(j-1))*T[i,j-1] - T[i-1,j-1])/((2^(j-1)) - 1)
       end
     end
+  else
+    let max_order=max_order, dt=dt, integrator=integrator, cache=cache, repeat_step=repeat_step,
+      uprev=uprev, T=T
+      Threads.@threads for i in 1:2
+        startIndex = (i==1) ? 1 : max_order
+        endIndex = (i==1) ? max_order-1 : max_order
+        for index in startIndex:endIndex
+          dt_temp = dt/(2^(index-1)) # Romberg sequence
+          W = calc_W!(integrator, cache, dt_temp, repeat_step)
+          k_copy = integrator.fsalfirst
+          u_tmp = uprev
+          for j in 1:2^(index-1)
+              k = _reshape(W\-_vec(dt_temp*k_copy), axes(uprev))
+              integrator.destats.nsolve += 1
+              u_tmp = u_tmp + k
+              k_copy = f(u_tmp, p, t+j*dt_temp)
+          end
+          T[index,1] = u_tmp
+        end
+      end
+
+      for i=2:max_order
+        for j=2:i
+          T[i,j] = ((2^(j-1))*T[i,j-1] - T[i-1,j-1])/((2^(j-1)) - 1)
```

Comment:
I guess, `max_order` is usually quite small, but still we should try to access `T` in memory order along columns to improve performance, see https://docs.julialang.org/en/v1.2/manual/performance-tips/#Access-arrays-in-memory-order,-along-columns-1.